### PR TITLE
Tmp issue 187

### PR DESCRIPTION
**Pull request by diptighosalkar**
Fixed FROMWSB block code.
Added alert for only TOWS_c simulation

### CODE REVIEW

File: SendLog.py [hunk 1]
```diff
@@ -946,10 +948,17 @@ def event_stream():
...
-            diagram.scilab_proc.poll() is not None:
+            diagram.scilab_proc.poll() is not None and \
+            diagram.workspace_counter != 1:
         print("log file is empty")
         yield "event: ERROR\ndata: log file is empty\n\n"
         return
+    # for Only TOWS_c block
+    if os.stat(diagram.log_name).st_size == 0 and \
+            diagram.workspace_counter == 1:
+        print("Variables is saved in workspace successfully")
```
Comment (sunilshetye) [inline SendLog.py]:
```suggestion
        print("Variables are saved in workspace successfully")
```

small fix

File: SendLog.py [hunk 2]
```diff
@@ -1530,6 +1539,41 @@ def run_scilab_func_expr_request():
...
+def run_scilab_func_str2code_request():
+    (__, __, scifile, sessiondir, __) = init_session()
+
+    file_name = join(sessiondir, "str2code_value.txt")
+    varnam = request.form['varnam']
+    '''
+    sample input to scilab:
+    varnam: aBc#12V
+    '''
+    command = "exec('" + FROMWSB_STR2CODE_SCI + \
```
Comment (sunilshetye) [inline SendLog.py]: Write this in two lines for readability. Also, use % on each line like this:

command = "exec('%s');" % FROMWSB_STR2CODE_SCI;
command += get_str2code_for_variablename('%s','%s');" % (filename, varname)

File: get_str2code_FROMWSB.sci [hunk 3]
```diff
@@ -0,0 +1,13 @@
+function get_str2code_for_variablename(filename,varnam)
+    f_temp = mopen(filename, 'wt');
+    variable = _str2code(varnam);
```
Comment (sunilshetye) [inline get_str2code_FROMWSB.sci]: We are already having _str2code in webapp/dependencies.js. Can't that be used directly?

File: webapp/data_structures/FROMWSB.js [hunk 4]
```diff
@@ -134,67 +134,99 @@ function FROMWSB() {
...
         }
-        this.x.model.ipar = new ScilabDouble([this.varnam.length],[_str2code(this.varnam)],[this.Method],[this.ZC],[this.OutEnd])
-        var io=set_io(this.x.model,this.x.graphics,[],[[-1],[-2]],[],[]);
-        var exprs=new ScilabString(this.varnam.toString(),this.Method,this.ZC,this.OutEnd)
-        this.x.graphics.exprs=exprs
+        this.varnam = varnam1;
+        this.Method = Method1;
+        this.ZC = ZC1;
+        this.OutEnd = OutEnd1;
+        var get_str2code = JSON.parse(get_str2code_output_for_set(this.varnam));
```
Comment (sunilshetye) [inline webapp/data_structures/FROMWSB.js]: Please check if we can use _str2code from webapp/dependencies.js here.

File: webapp/data_structures/FROMWS_c.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+function FROMWS_c () {
+
+    FROMWS_c.prototype.define = function FROMWS_c() {
+        this.varnam = "V";
+        this.Method = 1;
+        this.ZC = 1;
+        this.OutEnd = 0;
+        this.str2code = -31; //this is value which we get from _str2code(varnam) function.
```
Comment (sunilshetye) [inline webapp/data_structures/FROMWS_c.js]:
```suggestion
        this.str2code = _str2code("V");
```

Please check if we can do this.

File: SendLog.py (hunk 1 above)
Comment (diptighosalkar) [reply SendLog.py]: ok will modified stmt

File: SendLog.py (hunk 2 above)
Comment (diptighosalkar) [reply SendLog.py]: ok will change it as suggested

File: get_str2code_FROMWSB.sci (hunk 3 above)
Comment (diptighosalkar) [reply get_str2code_FROMWSB.sci]: ya i know about that function, I did study that before making this but what  _str2code in dependencies is returning is different from this one.

dependencies  _str2code is returning ascii value of string e.g. for 'abc' it return  97.    98.    99.
but scilab _strccode is returning some other number for 'abc' it return 10.  11.  12.  
Thus implemented this functionality

File: webapp/data_structures/FROMWSB.js [hunk 6]
```diff
@@ -53,7 +53,7 @@ function FROMWSB() {
                 state: new ScilabDouble(),
                 dstate: new ScilabDouble(),
                 odstate: list(),
-                rpar: new ScilabDouble(),
+                rpar: new ScilabDouble([]),
                 ipar: new ScilabDouble([1], [-31], [1], [1], [0]),
```
Comment (sunilshetye) [inline webapp/data_structures/FROMWSB.js]:
```suggestion
                ipar: new ScilabDouble([this.varnam.length],..._str2code(this.varnam),[this.Method],[this.ZC],[this.OutEnd]),
```

Suggestion for consistency

**Pull request merged**